### DISCUSSION TITLE
Why brew shows deprecation warning on pnpm@8 install?

### DISCUSSION THREAD

**Discussion by viking1304** [Homebrew/discussions#5356] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.21
ORIGIN: https://github.com/Homebrew/brew
HEAD: 82c2e743a5bcea725f9ca1429e3e21c3088ff904
Last commit: 5 days ago
Core tap JSON: 11 May 13:43 UTC
Core cask tap JSON: 11 May 13:43 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_EDITOR: code
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: dodeca-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.45.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I do not understand why brew shows deprecation warning on pnpm@8 install:

```
❯ brew install pnpm@8
==> Downloading https://formulae.brew.sh/api/formula.jws.json
##O#-  #
==> Downloading https://formulae.brew.sh/api/cask.jws.json
##O#-  #
Warning: pnpm@8 has been deprecated!
```

Disable date in formula is the correct one:

```
disable! date: "2025-04-30", because: :unmaintained
```

Is this related to https://github.com/endoflife-date/endoflife.date/pull/5062?

**Comment by SMillerDev** [comment]
> I do not understand why brew shows deprecation warning on pnpm@8 install

Because upstream released a new version and encouraged people to use it. Making the old version deprecated.

When they drop support it will be disabled completely

**Reply by viking1304** [reply]
> Because upstream released a new version and encouraged people to use it. Making the old version deprecated.

If that was some regular app I would totally agree with you, but with all due respect, in this case I disagree.

Most people only know that pnmp is another node package manager like npm and yarn, but they are not aware that pnpm can also replace nvm to manage different node versions.

If you check https://pnpm.io/installation#compatibility you will see that pnpm@9 supports node v18 and v20, but does not supports v16, adn pnpm@8 supports all 3 of them.

That means that with pnpm@8 users can switch between v16, v18 and v20 at any time, just like with nvm, but with pnpm@9 they can only use v18 and v20. So, pnpm@8 and pnpm@9 are more similar to for example python 3.11 and 3.12, than Outlook 2013 and 2016.

Can pnpm be used with nvm? Absolutely. I can also add deprecated formulas to my own tap at any time, so I have no problem. I am just trying to explain why I think that pnpm@8 should not be marked as deprecated as long as it is supported by devs.

**Reply by SMillerDev** [reply]
> pnpm@8 and pnpm@9 are more similar to for example python 3.11 and 3.12

They have a similar deprecation and disabling, they just have a longer time set by upstream.

**Reply by viking1304** [reply]
There is actually much better option for people who want to use pnpm as node version manager. There is a version of the pnpm CLI packaged with Node into an executable and it may be used even if Node is not installed.

I check what their [installer](https://github.com/pnpm/get.pnpm.io/blob/main/install.sh) does and I made a cask based on that installer. I will submit PR, but I guess it will require some more polishing.

**Reply by SMillerDev** [reply]
Seeing how it's an open source CLI tool that's already a formula: 
> App is both open-source and CLI-only (i.e. it only uses the binary artifact). In that case, and [in the spirit of deduplication](https://github.com/Homebrew/homebrew-cask/issues/15603), submit it first to [homebrew/core](https://github.com/Homebrew/homebrew-core) as a formula that builds from source. If it is rejected, you may then try again as a cask (link to the issue from your pull request so we can see the discussion and reasoning for rejection).

https://docs.brew.sh/Acceptable-Casks#rejected-casks

**Reply by viking1304** [reply]
I absolutely agree that CLI app should be formula if possible, and that there should be no duplication, but as you can see [here](https://pnpm.io/installation#using-npm) those are two different apps. Unlike "normal" pnpm, @pnpm/exe is an [actual macOS binary available for both ARM and x64](https://github.com/pnpm/pnpm/releases/). I would like to make a formula for @pnpm/exe but I have no idea how they make a macOS binary. I cannot find any build script or info how to build them. That was the reason I created the cask.
